git/repo/fun.py

```diff
@@ -208,7 +208,14 @@ def rev_parse(repo, rev):
                 ref = repo.head.ref
             else:
                 if token == '@':
-                    ref = name_to_object(repo, rev[:start], return_ref=True)
+                    try:
```

Comment:
A better solution might be to stop treating `@` as special outside of the spots that it is allowed in.
I am a hesitant to let it make assumptions unless there absolutely is no other way. Or in other words, I believe the `git` or `C` implementation is deterministic here

Reply:
Thanks for your feedback, I'll try it when I have time in the next weeks.